astropy/uncertainty/tests/test_distribution.py

```diff
@@ -316,3 +316,12 @@ def test_array_repr_latex():
 
     distr = Distribution(arr)
     assert distr._repr_latex_() is None
+
+
+def test_to():
+    distr = Distribution(np.random.randn(2, 100)*u.km)
+    distrm = distr.to(u.m)
+
+    assert distr.unit == u.km
+    assert distrm.unit == u.m
+    assert_quantity_allclose(distr.pdf_mean, distrm.pdf_mean)
```

Comment:
To check the class stays correct, could you add:
```
distr = Distribution(... * u.deg)
d_rad = distr.to(u.rad)
assert isinstance(d_rad, Angle)
d_one = distr.to(u.one, equivalencies=u.dimensionless_angles())
assert not isinstance(d_one, Angle)
assert isinstance(d_one, Quantity)
```
(typing without checking...)